sysconfig/cloudinit.go

```diff
@@ -76,21 +80,50 @@ func installCloudInitCfg(src, targetdir string) error {
 	return nil
 }
 
-// TODO:UC20: - allow cloud.conf coming from the gadget
-//            - think about if/what cloud-init means on "secured" models
+// installUnifiedCloudInitCfg installs a single cloud-init config file to the
+// cloud config dir as "ubuntu-core-cloud_91.cfg".
+func installUnifiedCloudInitCfg(src, targetdir string) error {
+	ubuntuDataCloudCfgDir := filepath.Join(ubuntuDataCloudDir(targetdir), "cloud.cfg.d/")
+	if err := os.MkdirAll(ubuntuDataCloudCfgDir, 0755); err != nil {
+		return fmt.Errorf("cannot make cloud config dir: %v", err)
+	}
+
+	configFile := filepath.Join(ubuntuDataCloudCfgDir, "80_device_gadget.cfg")
```

Comment:
see above, maybe the device_gadget portion of this name should come from outside? that's maybe just overkill and this function should be renamed ?

Reply:
I think for now we should just rename the function, eventually we should pass in the name of the file later depending on what file is being used, but that's a later refactor IMHO

Reply:
rename is fine